flink-connectors/flink-jdbc/src/main/java/org/apache/flink/api/java/io/jdbc/JDBCOutputFormat.java

```diff
@@ -160,7 +160,11 @@ public void writeRecord(Row row) throws IOException {
 									break;
 								case java.sql.Types.FLOAT:
 								case java.sql.Types.DOUBLE:
-									upload.setDouble(index + 1, (double) row.getField(index));
+									if (row.getField(index).getClass() == Float.class) {
```

Comment:
I am not sure this is the right approach. shouldn't we do something like:
```
case java.sql.Types.FLOAT:
  upload.setFloat(index + 1, (float) row.getField(index));
  break;
case java.sql.Types.DOUBLE:
  upload.setDouble(index + 1, (double) row.getField(index));
  break;
```
that would make more sense to me since the row is float, the type mapping is also float.

Reply:
agree